vald/sign.go

```diff
@@ -96,18 +114,44 @@ func GetSignCommand() *cobra.Command {
 			case *tofnd.SignResponse_Signature:
 				ecdsaSig := *funcs.Must(ec.ParseDERSignature(res.GetSignature()))
 				evmSignature := funcs.Must(evm.ToSignature(ecdsaSig, hash, pubKey.ToECDSAPubKey())).ToHomesteadSig()
-				fmt.Printf("signature: %s\n", hex.EncodeToString(evmSignature))
+
+				signDetails := map[string]string{
+					"key_id":    keyID.String(),
+					"validator": valAddr,
+					"msg_hash":  hex.EncodeToString(hash.Bytes()),
+					"pub_key":   pubKey.String(),
+					"signature": hex.EncodeToString(evmSignature),
+				}
+				fmt.Printf("%s", funcs.Must(json.MarshalIndent(signDetails, "", "  ")))
+
 				return nil
 			case *tofnd.SignResponse_Error:
 				return fmt.Errorf(res.GetError())
 			default:
 				panic(fmt.Errorf("unknown multisig sign response %T", res.GetSignResponse()))
 			}
-
 		},
 	}
 
-	cmd.Flags().String("validator-addr", "", "the address of the validator operator, i.e axelarvaloper1..")
+	cmd.Flags().String(flagPubKey, "", "the public key of the validator for the key id in hex format")
```

Comment:
What is the use case for this flag? Is it possible to have more than one public key associated with the key-id?

Reply:
nope, it gives the option to pass in the public key manually instead of retrieving it from a node (needed for certain scenarios, such as when the network is halted)

Reply:
queries should still work, so I don't think that's a valid scenario. Nevertheless, it's a useful flag to have, e.g. if you are running it on a machine that cannot reach the node